srv/adapter/claude.ts

```diff
@@ -59,36 +82,40 @@ export const handleClaude: ModelAdapter = async function* (opts) {
 
   log.debug(requestBody, 'Claude payload')
 
-  const resp = await needle('post', base.url, JSON.stringify(requestBody), {
-    json: true,
-    headers,
-  }).catch((err) => ({ error: err }))
+  const iterator = requestBody.stream
+    ? streamCompletion(base.url, requestBody, headers, opts.user._id, log)
+    : requestFullCompletion(base.url, requestBody, headers, opts.user._id, log)
+  let acc = ''
+  let resp: ClaudeCompletion | undefined
 
-  if ('error' in resp) {
-    log.error({ error: resp.error }, 'Claude request failed to send')
-    yield { error: `Claude request failed: ${resp.error?.message || resp.error}` }
-    return
-  }
+  while (true) {
+    let generated = await iterator.next()
 
-  if (resp.statusCode && resp.statusCode >= 400) {
-    log.error({ body: resp.body }, `Claude request failed (${resp.statusCode})`)
-    yield { error: `Claude request failed: ${resp.statusMessage}` }
-    return
+    if (generated.done) {
+      resp = generated.value
+      break
+    }
+
+    if ('error' in generated.value) {
+      yield generated.value
+      return
+    }
+
+    if ('token' in generated.value) {
+      acc += generated.value.token
+      yield {
+        partial: sanitiseAndTrim(acc, requestBody.prompt, opts.replyAs, opts.characters, members),
+      }
+    }
   }
 
   try {
-    const completion = resp.body.completion
+    const completion = resp?.completion || ''
     if (!completion) {
-      log.error({ body: resp.body }, 'OpenAI request failed: Empty response')
-      yield { error: `OpenAI request failed: Received empty response. Try again.` }
-      return
+      log.error({ body: resp }, 'Claude request failed: Empty response')
+      yield { error: `Claude request failed: Received empty response. Try again.` }
     } else {
-      const sanitised = sanitise(completion)
-      const trimmed = trimResponseV2(sanitised, opts.replyAs, members, opts.characters, [
-        'END_OF_DIALOG',
-      ])
-      yield trimmed || sanitised
-      return
+      yield sanitiseAndTrim(completion, requestBody.prompt, opts.replyAs, opts.characters, members)
```

Comment:
Tweaked to use `sanitiseAndTrim` from `common.ts` since I saw that's what the OpenAI adapter was already doing.

There is very subtle difference, though -- `sanitiseAndTrim` additionally removes the prompt from the returned text.  I don't think that would cause any problems here, but calling it out just in case that's a meaningful difference.

```
const parsed = sanitise(text.replace(prompt, ''))
```